daemon/execdriver/native/driver.go

```diff
@@ -198,7 +201,17 @@ func (d *Driver) Run(c *execdriver.Command, pipes *execdriver.Pipes, hooks execd
 	}
 	cont.Destroy()
 	destroyed = true
-	_, oomKill := <-oom
+	// Note: oomKilled will have an oom event if any process within the container
+	// died at any time, not only if the init process OOMed
+	// TODO, perhaps we only want the OOMKilled flag to be set if the OOM
+	// resulted in a container death (pid1 OOMd)
+	// Loop through all oomKilled values so we don't leak a goroutine in
```

Comment:
Is this actually true? I thought it might be after briefly skimming the codepath here, but I'm not completely sure.

I don't think there can be any harm in looping over all the channel's events anyways, but I wouldn't want to leave a misleading comment if that's not the case.